pkgs/build-support/fetchurl/builder.sh

```diff
@@ -128,6 +127,11 @@ set -o noglob
 
 success=
 for url in $urls; do
+    if [[ -z "$postFetch" ]] &&
+       [[ "$url" == https://github.com/*/archive/* ||
+          "$url" == https://gitlab.com/*/-/archive/* ]]; then
+        echo "warning: rev archives should use fetchzip instead"
```

Comment:
Might be better to specifically say something in the message that this is for when using fetchurl to fetch an archive from github and gitlab.

Reply:
This is not limited to GitHub or GitLab though: most rev archives (e.g. those generated by cgit, Gitea, Gitolite, and any other Git forge) have this property. So maybe it's better to match for `"$url" == https?://*/archive/*`.

I agree this message could be better, though. Do you have any ideas?

Reply:
It isn't generic to /archive/* though, so we can't give that generic advice: not all URLs with `/archive/` will offer a .zip. The message right now is also too generic: I don't what a rev archive is without looking at the source code of where the error came from. Since we do only warn on specific cases, we should offer specific feedback.

I would make the error message:

> warning: Archives from GitHub revisions should use fetchFromGitHub.

and:

> warning: Archives from GitLab revisions should use fetchFromGitLab.

and print the appropriate error depending upon which one is being fetched from.